core/src/main/scala/io/qbeast/core/transform/LinearTransformation.scala

```diff
@@ -57,12 +63,18 @@ case class LinearTransformation(minNumber: Any, maxNumber: Any, orderedDataType:
    * @param other
    * @return a new Transformation that contains both this and other.
    */
+  // TODO check if this is correct
   override def merge(other: Transformation): Transformation = {
     other match {
-      case LinearTransformation(otherMin, otherMax, otherOrdering)
+      case LinearTransformation(otherMin, otherMax, otherNullValue, otherOrdering)
           if orderedDataType == otherOrdering =>
-        LinearTransformation(min(minNumber, otherMin), max(maxNumber, otherMax), orderedDataType)
+        LinearTransformation(
+          min(minNumber, otherMin),
+          max(maxNumber, otherMax),
+          nullValue, // we should keep the original null value?
```

Comment:
We agree to choose otherNullValue since:

- If it's specified by the user, we should maintain the choice
- If it's a new null value, it would be adapted to the new space